shared/src/main/scala/mlscript/codegen/typescript/TsTypegen.scala

```diff
@@ -224,16 +242,18 @@ final case class TsTypegenCodeBuilder() {
         }
       case WithExtension(base, rcd) =>
         toTsType(Inter(Rem(base, rcd.fields.map(tup => tup._1)), rcd))
-      // type variables friendly names have been pre-calculated in the context
-      // throw an error if it doesn't exist
+      // get clash free name for type variable
       case t @ TypeVar(_, _) =>
-        print(typegenCtx.existingTypeVars)
-        typegenCtx.existingTypeVars
-          .get(t)
-          .map(s => SourceCode(s))
-          .getOrElse(
-            throw IllFormedTsTypeError(s"Did not find mapping for type variable $t. Unable to generated ts type.")
-          )
+        typegenCtx.findTypeVarName(t).fold(
+          name => SourceCode(name),
+          // type variable has been accessed for the first time
+          // allocate a clash free name in type variable mapping
+          friendlyName => {
+            val newName = typegenCtx.typeScope.allocateJavaScriptName(s"t$friendlyName")
```

Comment:
Why did we revert to using `t` prefixes?

Reply:
I saw cases like this pop-up.

```
//│ export declare const g<a>: (arg: [{a: a, b: int} & a]) => (arg1: [unknown]) => a
```

Record names having same identifiers as type parameters and becoming confusing. But now that I think of this is a contrived case, mostly because I'm writing toy tests with short names. This should not be a problem if records are defined well.

Still would it be good to have some convention for type parameters? Maybe using capital letters like, how mlscript type aliases need to start with captial letters.